server/service/externalsvc/zendesk.go

```diff
@@ -0,0 +1,155 @@
+package externalsvc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/fleetdm/fleet/v4/pkg/fleethttp"
+	"github.com/nukosuke/go-zendesk/zendesk"
+)
+
+// Zendesk is a Zendesk client to be used to make requests to the Zendesk external service.
+type Zendesk struct {
+	client  *zendesk.Client
+	groupID int64
+}
+
+// ZendeskOptions defines the options to configure a Zendesk client.
+type ZendeskOptions struct {
+	URL      string
+	Email    string
+	APIToken string
+	GroupID  int64
+}
+
+// NewZendeskClient returns a Zendesk client to use to make requests to the Zendesk external service.
+func NewZendeskClient(opts *ZendeskOptions) (*Zendesk, error) {
+	if os.Getenv("TEST_ZENDESK_CLIENT") == "true" {
+		return NewZendeskTestClient(opts)
+	}
+	client, err := zendesk.NewClient(fleethttp.NewClient())
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := url.Parse(opts.URL)
+	if err != nil {
+		return nil, err
+	}
+	subparts := strings.Split(url.Host, ".")
+	subdomain := subparts[0]
+
+	client.SetSubdomain(subdomain)
+	client.SetCredential(zendesk.NewAPITokenCredential(opts.Email, opts.APIToken))
+
+	return &Zendesk{
+		client:  client,
+		groupID: opts.GroupID,
+	}, nil
+}
+
+// GetGroup returns the group details for the group key provided in the
+// Zendesk client options. It can be used to test in one request the
+// authentication and connection parameters to the Zendesk instance as well as the
+// existence of the group.
+func (z *Zendesk) GetGroup(ctx context.Context) (*zendesk.Group, error) {
+	var group *zendesk.Group
+
+	op := func() (interface{}, error) {
+		g, err := z.client.GetGroup(ctx, z.groupID)
+		group = &g
+		return group, err
+	}
+
+	if err := doZendeskWithRetry(op); err != nil {
+		return nil, err
+	}
+	return group, nil
+}
+
+// CreateTicket creates a ticket on the Zendesk server targeted by the Zendesk client.
+// It returns the created ticket or an error.
+func (z *Zendesk) CreateTicket(ctx context.Context, ticket *zendesk.Ticket) (*zendesk.Ticket, error) {
+	ticket.GroupID = z.groupID
+
+	var createdTicket *zendesk.Ticket
+	op := func() (interface{}, error) {
+		t, err := z.client.CreateTicket(ctx, *ticket)
+		createdTicket = &t
+		return &createdTicket, err
+	}
+
+	if err := doZendeskWithRetry(op); err != nil {
+		return nil, err
+	}
+	return createdTicket, nil
+}
+
+func doZendeskWithRetry(fn func() (interface{}, error)) error {
```

Comment:
There's duplicated logic with https://github.com/fleetdm/fleet/blob/221232580c2151729756ed02e847c0f5c44c1e02/server/service/externalsvc/jira.go#L103-L143, maybe we can try to amend `doWithRetry` to re-use it?

Reply:
Leaving as TODO comment for the dev that implements the next integration is also a-ok :)